File: week_1/day-5-string-hashing/variations_que/palindromePermutation.js
function canPermutePalindrome(s) {
    const freq = {};
    for (let ch of s) freq[ch] = (freq[ch] || 0) + 1;

    let odd = 0;
    for (let key in freq) {
        if (freq[key] % 2 === 1) odd++;
        if (odd > 1) return false;
    }
    return true;
}

console.log(canPermutePalindrome("carrace")); // true

File: week_1/day-5-string-hashing/variations_que/palindrome_permutation.py
def canPermutePalindrome(s):
    freq = {}
    for ch in s:
        freq[ch] = freq.get(ch, 0) + 1

    odd_count = sum(1 for v in freq.values() if v % 2 == 1)
    return odd_count <= 1

# INPUT
print(canPermutePalindrome("carrace"))  # True
